backend/data_handling/data_processing.py

```diff
@@ -10,19 +10,26 @@
 """
 def clean_transaction_data(transaction_json):
     cleaned_data = []
-    for transaction in transaction_json['transactions']:
-        # filter out transactions where money is deposited to account since we're concerned with expenditures
-        if transaction["amount"] > 0 :
-            clean_transaction = {
-                # for the sake of the demo, amount spent on transactions is random to provide differeneces to analyze
-                "amount": round(random.uniform(0.01, 5000), 2),
-                "date": transaction["date"],
-                "currency": transaction["iso_currency_code"],
-                "detailed": transaction["personal_finance_category"]["detailed"],
-                "primary": transaction["personal_finance_category"]["primary"]
-            }
-            cleaned_data.append(clean_transaction)
-    return cleaned_data
+    # check to ensure each transaction has the expected fields, if not something is wrong with the data and we need to prevent a total crash
+    try :
+        for transaction in transaction_json['transactions']:
+                # filter out transactions where money is deposited to account since we're concerned with expenditures
+                if transaction["amount"] > 0 :
+                    clean_transaction = {
+                        # for the sake of the demo, amount spent on transactions is random to provide differeneces to analyze
+                        "amount": round(random.uniform(0.01, 5000), 2),
+                        "date": transaction["date"],
+                        "currency": transaction["iso_currency_code"],
+                        "detailed": transaction["personal_finance_category"]["detailed"],
+                        "primary": transaction["personal_finance_category"]["primary"]
+                    }
+                    cleaned_data.append(clean_transaction)
+                return cleaned_data
+    except Exception as e:
+        (f"Error cleaning transaction data: {e}")
```

Comment:
Currently this line evaluates the f string but does not do anything with it-- we should throw/log the error prior to returning

Reply:
Just fixed this in commit a365695